src/steamship/agents/schema/chathistory.py

```diff
@@ -272,3 +278,120 @@ def clear(self):
             self.embedding_index.reset()
 
         self.refresh()
+
+    def append_agent_message(
+        self,
+        text: str = None,
+        tags: List[Tag] = None,
+        content: Union[str, bytes] = None,
+        url: Optional[str] = None,
+        mime_type: Optional[MimeTypes] = None,
+    ) -> Block:
+        """Append a new block to this with status update messages from the Agent."""
+        return self.append_message_with_role(text, RoleTag.AGENT, tags, content, url, mime_type)
+
+    def append_tool_message(
+        self,
+        text: str = None,
+        tags: List[Tag] = None,
+        content: Union[str, bytes] = None,
+        url: Optional[str] = None,
+        mime_type: Optional[MimeTypes] = None,
+    ) -> Block:
+        """Append a new block to this with status update messages from the Agent."""
+        return self.append_message_with_role(text, RoleTag.TOOL, tags, content, url, mime_type)
+
+    def append_llm_message(
+        self,
+        text: str = None,
+        tags: List[Tag] = None,
+        content: Union[str, bytes] = None,
+        url: Optional[str] = None,
+        mime_type: Optional[MimeTypes] = None,
+    ) -> Block:
+        """Append a new block to this with status update messages from the Agent."""
+        return self.append_message_with_role(text, RoleTag.LLM, tags, content, url, mime_type)
+
+
+class ChatHistoryLoggingHandler(StreamHandler):
```

Comment:
I think it's really cool that we're hooking on to the logging system in this way.